service/controller/resource/instance/create_wait_for_nodes_to_become_ready.go

```diff
@@ -29,6 +31,31 @@ func (r *Resource) waitForWorkersToBecomeReadyTransition(ctx context.Context, ob
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", "found out that all tenant cluster worker nodes are Ready")
 
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
```

Comment:
I believe it would help understand the flow if we return `DeploymentUnintialized` instead of "".

Reply:
No, this returns error. See https://github.com/giantswarm/fmt/blob/master/go/errors.md#zero-values-when-returning-errors